File: ZMQ/lec-05-prg-05-pub-sub-and-pull-push-server.py
# Rerefence: https://stackoverflow.com/questions/49289072/zmq-dealer-router-communication

import zmq

def main():
    # context and sockets
    ctx = zmq.Context()
    publisher = ctx.socket(zmq.PUB)
    publisher.bind("tcp://*:5557")
    collector = ctx.socket(zmq.PULL)
    collector.bind("tcp://*:5558")

    while True:
        message = collector.recv()
        print("I: publishing update ", message)
        publisher.send(message)

if __name__ == '__main__':
    main()

File: ZMQ/lec-05-prg-05-pub-sub-and-pull-push-server.js
const zmq = require("zeromq");

async function run() {
  const publisher = new zmq.Publisher();
  await publisher.bind("tcp://*:5557");

  const collector = new zmq.Pull();
  await collector.bind("tcp://*:5558");

  while (true) {
    const message = await collector.receive();
    console.log("server: publishing update => ", message.toString());
    publisher.send(message.toString());
  }
}

run();
